cron/alerts/main.go

```diff
@@ -0,0 +1,330 @@
+}
+
+func getRecipientAddresses(a *AlertsConfig) error {
+	gluedRecipients := os.Getenv(EnvKeySESRecipients)
+	if gluedRecipients == "" {
+		return fmt.Errorf("required value missing for environment variable %s", EnvKeySESRecipients)
+	}
+
+	a.RecipientEmails = splitStringList(gluedRecipients)
+	return nil
+}
+
+func getCFContainsStrings(a *AlertsConfig) error {
+	gluedSearchStrings := os.Getenv(EnvKeyCFContainsStrings)
+	if gluedSearchStrings == "" {
+		return fmt.Errorf("required value missing for environment variable %s", EnvKeyCFContainsStrings)
+	}
+
+	a.CFContainsStrings = splitStringList(gluedSearchStrings)
+	return nil
+}
+
+func getRequiredString(envKey string, configEntry *string) error {
+	if *configEntry != "" {
+		return nil
+	}
+
+	value := os.Getenv(envKey)
+	if value == "" {
+		return fmt.Errorf("required value missing for environment variable %s", envKey)
+	}
+	*configEntry = value
+
+	return nil
+}
+
+func (a *AlertsConfig) init() error {
+
+	if err := getRequiredString(EnvKeyCFApiEmail, &a.CFApiEmail); err != nil {
+			return err
+		}
+
+	if err := getRequiredString(EnvKeyCFApiKey, &a.CFApiKey); err != nil {
+			return err
+	}
+
+	if len(a.CFContainsStrings) < 1 {
+		if err := getCFContainsStrings(a); err != nil {
+			return err
+		}
+	}
+
+	if len(a.CFZoneNames) < 1 {
+		if err := getZoneNames(a); err != nil {
+			return err
+		}
+	}
+
+	if len(a.RecipientEmails) < 1 {
+		if err := getRecipientAddresses(a); err != nil {
+			return err
+		}
+	}
+
+	if a.SESCharSet == "" {
+		a.SESCharSet = SESCharSet
+	}
+
+	if err := getRequiredString(EnvKeySESReturnToAddress, &a.SESReturnToAddr); err != nil {
+		return err
+	}
+
+	if a.SESAWSRegion == "" {
+		a.SESAWSRegion = getSESAWSRegion()
+	}
+
+	if err := getRequiredString(EnvKeySESSubject, &a.SESSubjectText); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+type AlertsConfig struct {
+	CFApiKey string `json:"CFApiKey"`
+	CFApiEmail string `json:"CFApiEmail"`
+	CFZoneNames      []string `json:"CFZoneNames"`
+	CFContainsStrings []string   `json:"CFContainsString"`
+	SESAWSRegion     string   `json:"SESAWSRegion"`
+	SESCharSet       string   `json:"SESCharSet"`
+	SESReturnToAddr  string   `json:"SESReturnToAddr"`
+	SESSubjectText   string   `json:"SESSubjectText"`
+	RecipientEmails  []string `json:"RecipientEmails"`
+}
+
+func getCFRecordsWithSubstring(substring, zoneName string, recs []cloudflare.DNSRecord, results map[string][]string) {
+	log.Printf("Records with '%s' in the name in zone: %s", substring, zoneName)
+
+	subRecs := []string{}
+	for _, r := range recs {
+		if len(r.Name) > 0 && strings.Contains(r.Name, substring) {
+			log.Print(" ", r.Name)
+			subRecs = append(subRecs, r.Name + " ... " + r.Content)
+		}
+	}
+	if len(subRecs) > 0  {
+		log.Print("-----")
+		results[zoneName] = subRecs
+	}
+
+}
+
+func getCFRecords(config AlertsConfig) (map[string][]string, error) {
+	api, err := cloudflare.New(config.CFApiKey, config.CFApiEmail)
+	if err != nil {
+		log.Fatal(err)
```

Comment:
Not in scope for this PR, but have you considered sending an email for this and other failures? This is a set-and-forget kind of app so notifying on error could be important.

Reply:
Good idea.  I'll try to get back to this.